workitem/expression_compiler.go

```diff
@@ -442,15 +448,16 @@ func (c *expressionCompiler) Child(e *criteria.ChildExpression) interface{} {
 
 	c.ensureJoinTable(tblName)
 	c.parameters = append(c.parameters, r)
-	c.parameters = append(c.parameters, r)
 
-	return fmt.Sprintf(`((iter.id = ?) OR replace("work_items".fields->>'%[1]s'::text, '-', '_') IN (
-		  SELECT replace(%[2]s.id::text, '-', '_')
-		  WHERE (%[2]s.path <@ (SELECT i.path
-		                        FROM %[3]s i
-						WHERE i.id = ? AND i.path <> ''
-					)
-						    )))`, left.FieldName, "iter", tblName)
+	return fmt.Sprintf(`(uuid("work_items".fields->>'%[1]s') IN (
+		SELECT i.id
+			FROM %[3]s i
+			WHERE
+				text2ltree(concat_ws('.', nullif(i.path, ''), replace(cast(i.id as text), '-', '_')))
```

Comment:
If this is the workaround for https://github.com/fabric8-services/fabric8-wit/issues/2203 then I would like to have https://github.com/fabric8-services/fabric8-wit/issues/2203 fixed first.